worktree_commit.go

```diff
@@ -61,16 +59,40 @@ func (w *Worktree) Commit(msg string, opts *CommitOptions) (plumbing.Hash, error
 		return plumbing.ZeroHash, err
 	}
 
+	allowEmpty := opts != nil && opts.AllowEmptyCommits
+
+	// First handle the case of the first commit in the repository being empty.
+	if len(opts.Parents) == 0 && len(idx.Entries) == 0 && !allowEmpty {
+		return plumbing.ZeroHash, ErrEmptyCommit
+	}
+
 	h := &buildTreeHelper{
 		fs: w.Filesystem,
 		s:  w.r.Storer,
 	}
 
-	treeHash, err = h.BuildTree(idx, opts)
+	treeHash, err := h.BuildTree(idx, opts)
 	if err != nil {
 		return plumbing.ZeroHash, err
 	}
 
+	previousTree := plumbing.ZeroHash
+	if len(opts.Parents) > 0 {
+		parentCommit, err := w.r.CommitObject(opts.Parents[0])
+		if err != nil {
+			// Parent might not be a commit.
+			if err != plumbing.ErrObjectNotFound {
+				return plumbing.ZeroHash, err
+			}
```

Comment:
I added these lines because sometimes parent was not a commit hash.

On `worktree_test.go:995` `TestStatusCheckedInBeforeIgnored`,

It threw error: `ErrObjectNotFound` when creating the first commit.

By inspecting opts.Parents[0], I got this hash which I was able to find what it is for: `6ecf0ef2c2dffb796033e5a02219af86ec6584e5`
```go
branch, err := r.Reference("refs/heads/master", false)
c.Assert(err, IsNil)
c.Assert(branch.Hash().String(), Equals, "6ecf0ef2c2dffb796033e5a02219af86ec6584e5")
```

I'm not sure if this is intended. But I added it so I can get any thoughts on it.

Reply:
The commit [6ecf0ef2c2dffb796033e5a02219af86ec6584e5](https://github.com/git-fixtures/basic/commit/6ecf0ef2c2dffb796033e5a02219af86ec6584e5) exists and should be a valid parent for a new commit when using the "Basic" repo fixture as source.

The test you referred to [is creating a fresh new repo](https://github.com/go-git/go-git/blob/cd6633c3c665ba37f766e96c59c57c58051d0e58/worktree_test.go#L991-L995), so the first commit would not have any parents. Maybe you are referring to another test that failed because of this?

Reply:
I found the problem causing this issue. You and I were both right.

Since `defaultTestCommitOptions` we use in the test is a pointer variable,
https://github.com/go-git/go-git/blob/cd6633c3c665ba37f766e96c59c57c58051d0e58/worktree_test.go#L37

It could be modified in other tests. 
Seems like what made the test fail is this code line down below.
https://github.com/go-git/go-git/blob/cd6633c3c665ba37f766e96c59c57c58051d0e58/options.go#L578

I'll change the `defaultTestCommitOptions` as function returning pointer variable.